ports/models.py

```diff
@@ -28,6 +33,236 @@ class Port(models.Model):
+                new_port.save()
+
+                try:
+                    new_port.categories.add(*port['categories'])
+                except KeyError:
+                    pass
+
+                try:
+                    for maintainer in port['maintainers']:
+                        name = maintainer.get('email', {}).get('name', '')
+                        domain = maintainer.get('email', {}).get('domain', '')
+                        github = maintainer.get('github', '')
+
+                        maintainer_object, created = Maintainer.objects.get_or_create(
+                            name=name,
+                            domain=domain,
+                            github=github
+                        )
+
+                        maintainer_object.ports.add(new_port)
+                except KeyError:
+                    pass
+
+                try:
+                    for variant in port['variants']:
+                        variant_object = Variant()
+                        variant_object.port = new_port
+                        variant_object.variant = variant
+                        variant_object.save()
+                except KeyError:
+                    pass
+
+        def load_dependencies_table(ports):
+
+            def load_depends(list_of_dependencies, type_of_dependency, port):
+                dependency = Dependency()
+                dependencies = []
+                dependency.type = type_of_dependency
+                dependency.port_name = port
+                for depends in list_of_dependencies:
+                    try:
+                        dependencies.append(Port.objects.get(name__iexact=depends.rsplit(':', 1)[-1]))
+                    except Port.DoesNotExist:
+                        print("Failed to append {} as a dependency to {}. Not Found.".format(depends.rsplit(':', 1)[-1],
+                                                                                             port.name))
+                dependency.save()
+                dependency.dependencies.add(*dependencies)
+
+            for port in ports:
+                try:
+                    port_object = Port.objects.get(name__iexact=port['name'])
+                    for dependency_type in ["lib", "extract", "run", "patch", "build", "test", "fetch"]:
+                        key = "depends_" + dependency_type
+                        if key in port:
+                            load_depends(port[key], dependency_type, port_object)
+
+                except Port.DoesNotExist:
+                    print("Failed to update dependencies for {}. Port not found in database.".format(port['name']))
+        if isinstance(data, list):
+            load_ports_and_maintainers_table(data)
+            load_categories_table(data)
+            load_dependencies_table(data)
+        elif isinstance(data, str):
+            ports = open_portindex_json(data)
+            load_categories_table(ports)
```

Comment:
Maybe have a small method for these 3 lines separately? We do not want to forget and repeat in case it supports a new format of input in future...

Reply:
Okay, thank you

Reply:
I think you probably forgot to push them or something. I just pulled in the latest code and I don't see the changes.

Reply:
I did the changes in: 22a718e .

Here is the the line from my fork: https://github.com/arjunsalyan/macports-gsoc-2019-webapp/blob/0c5f1491dc886ec530a3c947f6e4ac1839964d8f/ports/models.py#L255

Reply:
These two are not the same. The one here is part of `load` at around line 142 but the link above refers to changes in `update` around line 255.

Reply:
Uh Oh! I am sorry. I kept working on `updates`. Resolved.

Reply:
Thank you.

Reply:
I would have given it a different name though to keep the definition of `populate` consistent across other models like `BuildHistory`, `Submission`. Not a big deal, resolving the thread.